### PR TITLE
add timer for train to compare three DL frameworks' ability

### PR DESCRIPTION
**Pull request by StevenJokess**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1449/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1449/

**Comment by StevenJokess** [comment]
http://preview.d2l.ai/d2l-en/PR-1449/chapter_optimization/lr-scheduler.html
@astonzhang @mli @goldmermaid 
I have done some for the project.
astonzhang and mli never answer the discussion.
szha never reply me for two days.
The reason I messed around was your too slow actions and no any good reply and offer to me.
3 days ago, I was angry by my uncles' words that I was playing at home three months.
That moment I really want to die...
Now I was reported, but you should know you need to be quicker to catch my paces and give me anything that I can see hope for my life.
I really want to help you, but you just left me alone and let me down.
You deleted my helpful posts and all my issues reported.

**Comment by mli** [comment]
Thank you for sending the PR and sharing your thought. You are the most active member in the community to ask questions and suggestions in the past a few months. But I'd also like to share my thought:

1. We are maintaining the community in our volunteering time. We have other day jobs to do. Even for D2L, our major goal is publishing a high-quality printed version as soon as possible, given so many universities are using this book. Asking questions and suggestions do help us. But please do it in a professional way. In other words, we expect contributors use github issues/PR as writing an email, to make others easy to read and understand, instead of just chatting. I strongly encourage you to take an English class for better communication. 

2. I'm sorry to hear that your uncle judged you that "you was playing at home three months.". But please don't throw your emotions to others. You insulting replies that cursing other to die hurt others. You not only did it to d2l developers, and also to the mxnet community, and @soumith, the creator of pytorch, who kindly answered your question. This behavior is not tolerable. You even didn't apologize so far. 

3. In open source communities, people kindly help others. But don't think this kindness as a responsibility. You got all contents and helps for free, but it doesn't mean people should offer when you ask for more.

**Comment by astonzhang** [comment]
I am sorry to hear that you were feeling bad at that time. Everyone has down and up times. However, abusive comments hurt many others in the community. Though people may be kind to offer resources and help using their volunteering time and for free, it does not mean that they tolerate abusive behavior that clearly violates code of conduct. 

Getting others' help is a privilege, not a right. Everyone in the open-source community has priorities in their daily jobs and life, online and offline. They can be busy, out of bandwidth, stressed, and burned out. Thus, people may try to be responsive, but please bear with them if they have to focus on other priorities and help others.

P.S.: We didn't delete your issues: they disappear probably because your account was reported.

**Comment by astonzhang** [comment]
Thanks for the PR. We print out throughput for comparing different models. However, the LR scheduler does not introduce any new model, we may save space by hiding throughput info.

**Comment by StevenJokess** [comment]
@astonzhang 
In my opinion, "printing out throughput" can also compare three DL frameworks' ability, which is necessary to merge into a standard framework. It also keeps same output with tf2.

**Comment by StevenJokess** [comment]
@mli 
Thank you for the reply and sharing your thought. 
1. I know that "We are maintaining the community in our volunteering time. We have other day jobs to do.", but community also does need your actions to make it more active. I don't think you are publishing D2L "ASAP". I added some discussions' URLs to try to help you, and you just left me alone. And I also know that my English is bad and I need to learn it to be better. But now it is also not my priority. How to get money is only important thing to survive, then I'll have rights to require teacher's feedback to my English class, as you said. 
2. To die can be a bless for whom don't have any hope for their life and just feel suffering. I apologize for all people if you think it hurt you, but I also envy you guys when you said it is a curse. 
3. You are also right. I never think this kindness as a responsibility. That is also why I tried "Die!" words to get your attention. I have no right and money to require you. All I have is my unimportance fame, so I tolerance a bad fame to get your attention.
How many times I begged you for a "product manager" job to help you and learn at the same time.
It never be replied. You need to think as a real "hungry" freshman when you are trying to teach others and take more attention to questions in forum.

**Comment by mli** [comment]
Job d2l-en/PR-1449/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1449/

**Comment by astonzhang** [comment]
Comparing frameworks' performance is not our goal though this can be done by users. We want to keep code simple.

**Comment by mli** [comment]
> How to get money is only important thing to survive

Then you should focus on it. Contributing to D2L will not get you money. We
expect our contributors have done their primary jobs before helping us. So
please find a job first. I know this year is hard, at least my team is not
hiring full-time in China right now. You should have 100% of your time on
searching jobs. We'll welcome you back if you settle on your new job.

**Comment by StevenJokess** [comment]
@mli @astonzhang  
Thanks to your reply very much.
It would be very helpful to find a job if you add my picture to **Pytorch Adaptation** (translated chapter_generative-adversarial-networks) in index.html.
Please help me.
I will contribute more if I find a job.

**Comment by mli** [comment]
We will list you in the preface to acknowledge your contribution. Here is
the paragraph in the contribution guideline about authorship:

For adapting D2L code into other frameworks, contributors who ported D2L
code of at least 40 entire notebooks will be listed on the D2L index page
as “XXX Adaptation Authors” (XXX can be PyTorch, TensorFlow, etc.). The
rest of contributors will be acknowledged in the preface
<https://github.com/d2l-ai/d2l-en/blob/master/chapter_preface/index.md#acknowledgments>.
Please add yourself if we missed it.

**Comment by StevenJokess** [comment]
@mli @astonzhang 
There are not enough 40 notebooks left...
And the difficulty of last chapter(
http://preview.d2l.ai/d2l-en/master/chapter_generative-adversarial-networks/gan.html
http://preview.d2l.ai/d2l-en/master/chapter_generative-adversarial-networks/dcgan.html)
is more challenging than others in general. It is more like a real project.

Besides,
I have 
1. fixed lots of typo (something like "title = Dive into Deep Learing")
2. added helpful discussions 
(https://discuss.d2l.ai/t/do-these-before-you-ask/540/5 
https://discuss.d2l.ai/t/using-jupyter/421/2
...
Too many
You have time to **look through** the whole book's discussion then you will find that I am not bullshiting...)
image
3. 20+ issues(all disappear... You should take attention before.)
4. replied a lot of questions.
5. share my reading repo: https://github.com/StevenJokess/d2l-en-read

You can find some contributions: https://github.com/dltech-xyz/d2l-en-read/tree/master/My%20contribution
image

I'm hunting a job to start AI.(I survive just for AI. Now I can't find any meaning for my life.)
It is really important for my whole life.
I don't have a famous or master degree to prove.
It is all I have.
Please help me...

**Comment by astonzhang** [comment]
Closing this PR due to lack of relevant activities.
